### PR TITLE
Adding Music On Console (mocp) support, requires >= 2.3.0 version

### PR DESCRIPTION
**Pull request by gsusrafael**

### CODE REVIEW

File: powerline/segments/common/players.py [hunk 1]
```diff
@@ -528,3 +528,61 @@ def get_player_status(self, pl):
 
 {0}
 ''').format(_common_args.format('itunes')))
+
+
+class MocPlayerSegment(PlayerSegment):
+	def get_player_status(self, pl):
+		'''Return Music On Console (mocp) player information.
+
+		mocp -i returns current information i.e.
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: 1. Missing double backticks around `mocp -i`.
2. Below is not how code is formatted in rst, you need

```ReST
…

.. code-block::

   <code here, I align it with first `c` from `code-block`>

…
```

File: powerline/segments/common/players.py [hunk 2]
```diff
@@ -528,3 +528,61 @@ def get_player_status(self, pl):
...
+		ignore_info and brings the key inside that to the right info of
+		the dictionary.
+		'''
+		now_playing_str = run_cmd(pl, ['mocp', '-i'])
+		if not now_playing_str:
+			return
+		ignore_info = (
+		    'File', 'TimeLeft', 'Title',
+		    'Bitrate', 'AvgBitrate', 'Rate'
+		)
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: Remove `ignore_info` completely, it has no purpose.

File: powerline/segments/common/players.py [hunk 3]
```diff
@@ -528,3 +528,61 @@ def get_player_status(self, pl):
...
+		now_playing = dict(
+		    [line.split(': ')
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: Use

```
dict((
    line.split(': ')
    for line in now_playing_str.split('\n')[:-1]
))
```

File: powerline/segments/common/players.py [hunk 4]
```diff
@@ -528,3 +528,61 @@ def get_player_status(self, pl):
...
+			'title': now_playing.get('SongTitle'),
+			'elapsed': _convert_seconds(now_playing.get('CurrentSec', 0)),
+			'total': _convert_seconds(now_playing.get('TotalSec', 0)),
+		}
+
+
+mocp = with_docstring(MocPlayerSegment(),
+('''Return MOC (Music On Console) player information
+
+Requires mocp command be acessible from $PATH.
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: If there are known version requirements it is best to put them to documentation.

**Comment by ZyX-I** [comment]
`ignore_info` is still present.

File: powerline/segments/common/players.py [hunk 5]
```diff
@@ -528,3 +528,62 @@ def get_player_status(self, pl):
...
 {0}
 ''').format(_common_args.format('itunes')))
+
+
+class MocPlayerSegment(PlayerSegment):
+	def get_player_status(self, pl):
+		'''Return Music On Console (mocp) player information.
+
+		`mocp -i` returns current information i.e.
+		.. code-block::
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: Missing empty lines around `.. code-block::`.

**Comment by gsusrafael** [comment]
Now ignore_info makes sense... check again the code


On Tue, Aug 01, 2017 at 04:30:17AM -0700, Nikolai Aleksandrovich Pavlov wrote:
> `ignore_info` is still present.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/powerline/powerline/pull/1811#issuecomment-319344536

-- 
Jesus Rafael Sanchez 
"Inter fractal umbras et illusionibus evanuit somnia ..."

**Comment by ZyX-I** [comment]
@gsusrafael It does not. You are not passing the dictionary anywhere so no need to filter out anything when constructing it. With double line splitting and O(N) lookup in `ignore_info` it is not even an optimization. Remove.

**Comment by ZyX-I** [comment]
And do not forget to remove `ignore_info` from doc as well.

File: powerline/segments/common/players.py [hunk 6]
```diff
@@ -528,3 +528,58 @@ def get_player_status(self, pl):
...
+		dictionary comprehension in this method takes anything in
+		ignore_info and brings the key inside that to the right info of
+		the dictionary.
+		'''
+		now_playing_str = run_cmd(pl, ['mocp', '-i'])
+		if not now_playing_str:
+			return
+
+		now_playing = dict((
+                    line.split(': ', 1)
```
Comment (ZyX-I) [inline powerline/segments/common/players.py]: Looks like if you have spaces here.

**Pull request merged**